src/app/ui/dialog/dialog.component.ts

```diff
@@ -1,12 +1,28 @@
-import { ChangeDetectionStrategy, Component, ElementRef, type OnInit, output, viewChild } from '@angular/core';
+import { NgClass, NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  TemplateRef,
+  contentChild,
+  effect,
+  input,
+  output,
+  viewChild,
+} from '@angular/core';
 
 import { IconComponent } from '../icon/icon';
 
 @Component({
   selector: 'app-dialog',
   template: `
     <dialog
-      class="m-0 h-full w-full bg-primary p-5 pt-16 text-primary [max-block-size:unset] [max-inline-size:unset]"
+      [ngClass]="[
+        'fixed top-0 grid h-full w-full bg-primary p-5 pt-16 text-primary [max-block-size:unset] [max-inline-size:unset]',
+        '[&:not([open])]:pointer-events-none',
+        'transition-transform duration-300 [&:not([open])]:translate-x-full',
```

Comment:
couldn't figure out fast enough how native dialog can work with angular animations and with `@if {}` condition around dialog component to get rid of `@if` + `ng-template` for a content